packages/flutter/lib/src/material/input_date_picker_form_field.dart

```diff
@@ -195,25 +214,25 @@ class _InputDatePickerFormFieldState extends State<InputDatePickerFormField> {
     return null;
   }
 
+  bool _updateDate(String? text) {
+    final DateTime? date = _parseDate(text);
+    if (_isValidAcceptableDate(date)) {
+      _selectedDate = date;
+      _inputText = text;
+      return true;
+    }
+    return false;
+  }
+
   void _handleSaved(String? text) {
-    if (widget.onDateSaved != null) {
-      final DateTime? date = _parseDate(text);
-      if (_isValidAcceptableDate(date)) {
-        _selectedDate = date;
-        _inputText = text;
-        widget.onDateSaved!(date!);
-      }
+    if (_updateDate(text)) {
```

Comment:
It's a bit odd that this helper function returns a bool but also modifies properties. I wouldn't expect a helper called this way to have those side effects